gax/BUILD.bazel

```diff
@@ -19,17 +19,21 @@ licenses(["notice"])  # Apache 2.0
 cc_library(
     name = "gax",
     srcs = [
+        "backoff_policy.cc",
         "status.cc",
     ],
     hdrs = [
+        "backoff_policy.h",
         "retry_policy.h",
         "status.h",
     ],
     deps = [
+        "@gtest//:gtest",
```

Comment:
Update: if we are going to use the FRIEND_TEST macro defined in gtest_prod.h, we should depend only on gtest_prod here (see BUILD.bazel file: https://github.com/google/googletest/blob/master/BUILD.bazel#L51)

But I'd also like to get your opinion on whether it is a good idea to test private members in this way via friend classes - I'm unsure about best practice. WDYT?

Reply:
You're right, that is the correct thing to do.

The version of gtest that we're depending on doesn't seem to provide a separate gtest_prod target, or at least the build failed when I tried to use it. The header exists, but it's subsumed by the gtest target. I'll double check though, because I may have been doing something silly.

Reply:
SGTM, if necessary we can bump up the version we depend on.

Reply:
Double check confirms the error and root cause I was seeing previously:
`no such target '@gtest//:gtest_prod': target 'gtest_prod' not declared in package`

Downloading and untarring gtest-1.8.1, the version labeled in repositories.bzl, confirms that there are only 'gtest' and 'gtest_main' targets.

To address your above question as to whether it's even a good idea to test private members like this via friends, I would say that it is. The ownership and construction of the rng is an implementation detail, but its one with serious although mostly client-transparent impact.
As for the current_delay_rang_ and maximum_delay_, we want to make strong guarantees about the class invariants that the public interface doesn't let us test directly. It's completely possible to go overboard with friendly access, but I don't think that's the case here.

Reply:
:-(

Here is the definition of FRIEND_TEST, it is a one-liner: https://github.com/google/googletest/blob/master/googletest/include/gtest/gtest_prod.h#L59

I would prefer to either (a) define the macro ourselves or (b) just explicitly write out `friend class ...` than take a dependency on gtest in the main library.

Reply:
Define the macro ourselves it is.